src/App.tsx

```diff
@@ -1,14 +1,65 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import './App.scss';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedFilter, setCompletedFilter] = useState('all');
+  const [isTodoLoading, setIsTodoLoading] = useState(false);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+
+  useEffect(() => {
+    setIsTodoLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setIsTodoLoading(false));
+  }, []);
+
+  const selectedTodo = useMemo(() => {
+    return todos.find((todo) => todo.id === selectedTodoId);
+  }, [selectedTodoId, todos]);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      const { title, completed } = todo;
+      const preparedTitle = title.toLowerCase().trim();
+      const preparedQuery = searchQuery.toLowerCase().trim();
+
+      const filterByTitle = preparedTitle.includes(preparedQuery);
+
+      switch (completedFilter) {
+        case 'active':
+          return filterByTitle && !completed;
+
+          break;
+        case 'completed':
+          return filterByTitle && completed;
+          break;
+
+        case 'all':
+        default:
+          return filterByTitle;
+          break;
+      }
+    });
```

Comment:
it would be better to move filter logic to helper function

Reply:
Agree